mmv1/products/firebase/api.yaml

```diff
@@ -152,3 +152,61 @@ objects:
           Immutable. The globally unique, Firebase-assigned identifier of the App.
 
           This identifier should be treated as an opaque token, as the data format is not specified.
+  - !ruby/object:Api::Resource
+    name: 'AndroidApp'
+    min_version: beta
+    base_url: projects/{{project}}/androidApps
+    self_link: '{{name}}'
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      A Google Cloud Firebase Android application instance
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://firebase.google.com/'
+      api: 'https://firebase.google.com/docs/projects/api/reference/rest/v1beta1/projects.androidApps'
+    async: !ruby/object:Api::OpAsync
+      actions: ["create"]
+      operation: !ruby/object:Api::OpAsync::Operation
+        path: 'name'
+        base_url: '{{op_id}}'
+        wait_ms: 1000
+      result: !ruby/object:Api::OpAsync::Result
+        path: 'response'
+        resource_inside_response: true
+      status: !ruby/object:Api::OpAsync::Status
+        path: 'done'
+        complete: true
+        allowed:
+          - true
+          - false
+      error: !ruby/object:Api::OpAsync::Error
+        path: 'error'
+        message: 'message'
+    properties:
+      - !ruby/object:Api::Type::String
+        name: name
+        description: |
+          The fully qualified resource name of the App, for example:
+
+          projects/projectId/androidApps/appId
+        output: true
+      - !ruby/object:Api::Type::String
+        name: displayName
```

Comment:
as the only user-defined field, have you tested the ability to update `displayName` ?

Reply:
We only create the resource in this PR, during which displayName gets set. Update resource will probably be added in a subsequent PR.

Reply:
Testing update functionality cannot be split out into a separate PR, if this PR is merged the resource may be released to the public without your subsequent PR having also been merged first.

All that really needs to happen is manually updating the `display_name` in a local provider and verifying that it works, or creating a new handwritten test that updates the `display_name` in the test itself.

Reply:
Hey sorry about being unclear. You have two options: create a handwritten test that updates the field, or [manually testing with a local provider binary](https://github.com/GoogleCloudPlatform/magic-modules#using-released-terraform-binary-with-local-provider-binary).

to create a handwritten test, you'll need an entirely new `.go` file that is separate from the `.tf.erb` example files. [See this file](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/tests/resource_cloud_iot_device_update_test.go) for an example of a standard handwritten update test.

Reply:
@ c2thorn Added a handwritten test. PTAL. Thanks.